docs/threat-model.md

```diff
@@ -0,0 +1,54 @@
+
+#### Lab
+
+Some netperf tests are run on machines in a physical lab ([details](machines#lab-x64-machines)). The lab machines do not have access to any corpnet resources. They have internet access and are on a shared, private network with each other (used for the network testing). The machines are controlled by the netperf framework and are used to run the tests.
+
+#### Dependencies
+
+netperf has dependencies on various other repositories which contain the tools and libraries under test. These dependencies are used to build the tools and libraries and are used in the tests.
+
+## Threat Surface
+
+The following section will focus on threats exposed via netperf itself, and will not cover things like physical security of the lab, or the security of the Azure Subscription beyond what netperf itself exposes.
+
+### Pull Requests
+
+Pull Requests are a primary attack vector for netperf. Pull Requests can trigger the netperf automation to run code to access the Azure Subscription and lab machines. If left unsecured, this could allow an attacker to run arbitrary code on the machines in the lab or in Azure, as well as leverage the Azure Subscription for their own purposes.
+
+This threat is largely mitigated by the following:
+
+1. **Approval Required**: Pull Requests from outside contributors are not automatically run. They must be approved by a netperf maintainer before they will be run. All Pull Requests from outside contributors are thouroughly reviewed before approval.
+2. **Secrets**: Secrets are stored in GitHub Secrets and are not exposed in the Pull Request. This means that an attacker would need to compromise the GitHub repository to access the secrets.
+
+Additionally, the Federated credentials that are used to access the Azure Subscription are limited in scope and only allowed to be executed from the main branch of the netperf repository. So, even if a Pull Request were to be run, and somehow get access to the secrets, it would not have access to the Azure Subscription.
+
+### Remote Action Triggering
+
+Since netperf is a framework for other repositories to use, netperf is setup to allow those repositories to remotely trigger netperf's GitHub actions. They do this via [workflow_call](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_call). A helper has been explicitly created, and you may find more details [here](arch#usage).
+
+The threat of unauthorized access to triggering of the actions is mitigated via general GitHub repository security. Triggers of the GitHub Actions requires write priviliges to netperf. To achieve this requires a token to securely access the netperf repository. In the other repositories, this token is stored in GitHub Secrets and is not directly exposed in the repository, nor to outside Pull Requests. This means that an attacker would need to compromise the repository to access the token. This prevents unauthorized access to the netperf repository.
+
+### Dependencies
```

Comment:
Dependencies should be pinned by hash. Either a git commit id if the dependency is built or an explicit hash of the file if it is a script or binary.

Reply:
I think it's worth documenting that all GitHub actions and dependent binaries / scripts must be pinned by hash. They are today, but it's a potential supply chain attack that we need to mitigate.